### PR TITLE
E2e salesforce datavalidation

### PR DESCRIPTION
**Pull request by AnkitCLI**

### CODE REVIEW

File: src/e2e-test/features/salesforcebatchsource/RunTime.feature [hunk 1]
```diff
@@ -27,7 +27,7 @@ Feature: Salesforce Batch Source - Run time Scenarios
     And Navigate to the properties page of plugin: "Salesforce"
     And fill Authentication properties for Salesforce Admin user
     And Enter input plugin property: "referenceName" with value: "Reference"
-    And Enter input plugin property: "sObjectName" with value: "sobject.lead"
+    And Enter input plugin property: "sObjectName" with value: "automation"
```
Comment (Vipinofficial11) [inline src/e2e-test/features/salesforcebatchsource/RunTime.feature]: Instead of simply writing ```automation```  write ```sobject.Automation_custom_c``` for a better readability

File: src/e2e-test/features/salesforcebatchsource/RunTime.feature [hunk 2]
```diff
@@ -126,3 +126,42 @@ Feature: Salesforce Batch Source - Run time Scenarios
     And Open and capture logs
     And Verify the pipeline status is "Succeeded"
     Then Close the pipeline logs
+
+  @BATCH-TS-SF-RNTM-04 @BQ_SINK_TEST @BQ_TEMP_CLEANUP @CREATE_CUSTOM @DELETE_OBJECT
```
Comment (Vipinofficial11) [inline src/e2e-test/features/salesforcebatchsource/RunTime.feature]: There are test cases above which are already running pipeline for salesforce to BQ no need to write a new tests. Add the new hooks and data validation to the above tests.

File: src/e2e-test/features/salesforcemultiobjectsbatchsource/RunTime.feature [hunk 3]
```diff
@@ -131,4 +131,46 @@ Feature: Salesforce Multi Objects Batch Source - Run time Scenarios
     And Wait till pipeline is in running state
     And Open and capture logs
     And Verify the pipeline status is "Succeeded"
-    Then Close the pipeline logs
\ No newline at end of file
+    Then Close the pipeline logs
+
+
+
+  @MULTIBATCH-TS-SF-RNTM-04 @BQ_SINK_TEST @BQ_TEMP_CLEANUP @CREATE_CUSTOM @DELETE_OBJECT
```
Comment (Vipinofficial11) [inline src/e2e-test/features/salesforcemultiobjectsbatchsource/RunTime.feature]: Same thing here update the above tests with data validation logic no need to add a new test, also the object are same.

File: src/e2e-test/features/salesforcesink/RunTime.feature [hunk 4]
```diff
@@ -126,4 +126,42 @@ Feature: Salesforce Sink - Run time Scenarios
     Then Wait till pipeline is in running state
     Then Open and capture logs
     Then Verify the pipeline status is "Succeeded"
-    Then Close the pipeline logs
\ No newline at end of file
+    Then Close the pipeline logs
+
+  @SINK-TS-SF-RNTM-05 @BQ_SOURCE_TEST @DELETE_OBJECT
+  Scenario: To verify data is getting transferred from Big Query source to Salesforce sink successfully
```
Comment (Vipinofficial11) [inline src/e2e-test/features/salesforcesink/RunTime.feature]: Same comment update the above tests and remove this one.

File: src/e2e-test/java/io/cdap/plugin/BQValidation.java [hunk 5]
```diff
@@ -0,0 +1,231 @@
...
+ */
+public class BQValidation {
+
+  static List<JsonObject> bigQueryResponse = new ArrayList<>();
+  static List<Object> bigQueryRows = new ArrayList<>();
+  static Gson gson = new Gson();
+
+  public static boolean validateSalesforceToBQRecordValues(String objectName, String targetTable) throws
+    IOException, InterruptedException {
+    String id = SalesforceClient.id;
```
Comment (Vipinofficial11) [inline src/e2e-test/java/io/cdap/plugin/BQValidation.java]: change it from ```id``` to ```uniqueRecordId```.

File: src/e2e-test/java/io/cdap/plugin/BQValidation.java [hunk 6]
```diff
@@ -0,0 +1,231 @@
...
+    for (Object rows : bigQueryRows) {
+      JsonObject jsonData = gson.fromJson(String.valueOf(rows), JsonObject.class);
+      bigQueryResponse.add(jsonData);
+    }
+    SalesforceClient.queryObject(id, objectName);
+    return compareSalesforceAndJsonData(SalesforceClient.objectResponseList, bigQueryResponse, targetTable);
+
+  }
+
+  public static boolean validateSalesforceMultiObjectToBQ(String objectName) throws IOException,
```
Comment (Vipinofficial11) [inline src/e2e-test/java/io/cdap/plugin/BQValidation.java]: Update this logic as it won't work for 2 different Sobjects

File: src/e2e-test/java/io/cdap/plugin/BQValidation.java [hunk 7]
```diff
@@ -0,0 +1,231 @@
...
+  public static TableResult getTableNamesFromDataSet() throws IOException, InterruptedException {
+    String projectId = PluginPropertyUtils.pluginProp("projectId");
+    String dataset = PluginPropertyUtils.pluginProp("dataset");
+    String selectQuery = "SELECT table_name FROM `" + projectId + "." + dataset + "`.INFORMATION_SCHEMA.TABLES ";
+
+    return BigQueryClient.getQueryResult(selectQuery);
+  }
+
+  public static String getTableByName() throws IOException, InterruptedException {
+    String tableName = PluginPropertyUtils.pluginProp("bqTargetTable");
```
Comment (Vipinofficial11) [inline src/e2e-test/java/io/cdap/plugin/BQValidation.java]: I believe this method is used for multi object validation. Try to handle few edge cases here. 

1. Check if table name is same as Object name, if yes add those 2 tables names in the list .In case your list don't have any table fail the test here only with a message that database don't have any table or something.

2. if you are using 2 tables in the source you need to 2 tables from the list, so check for this size once you have retrived the tables if size is not same fail the test.

File: src/e2e-test/java/io/cdap/plugin/BQValidation.java [hunk 8]
```diff
@@ -0,0 +1,231 @@
...
+
+  public static boolean compareSalesforceAndJsonData(List<JsonObject> salesforceData, List<JsonObject> bigQueryData,
+                                                     String tableName) {
+    boolean result = false;
+    if (bigQueryData == null) {
+      Assert.fail("bigQueryData is null");
+      return result;
+    }
+
+    int columnCountTarget = 0;
```
Comment (Vipinofficial11) [inline src/e2e-test/java/io/cdap/plugin/BQValidation.java:139]: If this is not used you can remove it.

File: src/e2e-test/java/io/cdap/plugin/BQValidation.java [hunk 9]
```diff
@@ -0,0 +1,231 @@
...
+  }
+
+  public static boolean compareSalesforceAndJsonData(List<JsonObject> salesforceData, List<JsonObject> bigQueryData,
+                                                     String tableName) {
+    boolean result = false;
+    if (bigQueryData == null) {
+      Assert.fail("bigQueryData is null");
+      return result;
+    }
+
```
Comment (Vipinofficial11) [inline src/e2e-test/java/io/cdap/plugin/BQValidation.java:157]: You may have to add a edge case to check the row count of both source and sink

File: src/e2e-test/java/io/cdap/plugin/BQValidation.java [hunk 10]
```diff
@@ -0,0 +1,231 @@
...
+    // Get the table schema
+    Schema schema = bigQuery.getTable(tableRef).getDefinition().getSchema();
+
+    // Iterate over the fields
+    int currentColumnCount = 1;
+    while (currentColumnCount <= columnCountSource) {
+      for (Field field : schema.getFields()) {
+        String columnName = field.getName();
+        String columnType = field.getType().toString();
+        String columnTypeName = field.getType().getStandardType().name();
```
Comment (Vipinofficial11) [inline src/e2e-test/java/io/cdap/plugin/BQValidation.java]: Its unused I believe, remove it if its not used.

File: src/e2e-test/java/io/cdap/plugin/BQValidation.java [hunk 11]
```diff
@@ -0,0 +1,231 @@
...
+          case "FLOAT":
+            Double sourceVal = salesforceData.get(0).get(columnName).getAsDouble();
+            Double targetVal = bigQueryData.get(jsonObjectIdx).get(columnName).getAsDouble();
+            Assert.assertEquals(String.format("Different values found for column: %s", columnName), 0,
+                                sourceVal.compareTo(targetVal));
+            break;
+
+          case "TIMESTAMP":
+            OffsetDateTime sourceTimestamp = OffsetDateTime.parse(salesforceData.get(0).get(columnName).getAsString(),
+                                                                  DateTimeFormatter.ofPattern(
```
Comment (Vipinofficial11) [inline src/e2e-test/java/io/cdap/plugin/BQValidation.java]: Correct the code formating in this method. For ex :  (2 space indentation)

File: src/e2e-test/java/io/cdap/plugin/package-info.java [hunk 12]
```diff
@@ -0,0 +1,20 @@
+/*
+ * Copyright © 2023 Cask Data, Inc.
```
Comment (Vipinofficial11) [inline src/e2e-test/java/io/cdap/plugin/package-info.java:2]: What is this file being used for ?

File: src/e2e-test/java/io/cdap/plugin/salesforcebatchsource/stepsdesign/DesignTimeSteps.java [hunk 13]
```diff
@@ -76,4 +88,35 @@ public void clickOnNewServiceNowConnection() {
...
+  @Then("Validate the values of records transferred to target Big Query table is equal to the values from source table")
+  public void validateTheValuesOfRecordsTransferredToTargetBigQueryTableIsEqualToTheValuesFromSourceTable()
+    throws IOException, InterruptedException, IOException, SQLException, ClassNotFoundException, ParseException {
+    int targetBQRecordsCount = BigQueryClient.countBqQuery(PluginPropertyUtils.pluginProp("bqTargetTable"));
+    BeforeActions.scenario.write("No of Records Transferred to BigQuery:" + targetBQRecordsCount);
+    Assert.assertEquals("Out records should match with target BigQuery table records count",
+                        CdfPipelineRunAction.getCountDisplayedOnSourcePluginAsRecordsOut(), targetBQRecordsCount);
+
+    boolean recordsMatched = BQValidation.validateSalesforceToBQRecordValues
+      (PluginPropertyUtils.pluginProp("customObject"),
```
Comment (Vipinofficial11) [inline src/e2e-test/java/io/cdap/plugin/salesforcebatchsource/stepsdesign/DesignTimeSteps.java]: fix the identation/spacing

File: src/e2e-test/java/io/cdap/plugin/salesforcebatchsource/stepsdesign/DesignTimeSteps.java [hunk 14]
```diff
@@ -76,4 +88,35 @@ public void clickOnNewServiceNowConnection() {
...
+       PluginPropertyUtils.pluginProp("bqTargetTable"));
+    Assert.assertTrue("Value of records transferred to the target table should be equal to the value " +
+                        "of the records in the source table", recordsMatched);
+  }
+  @Then("Read Credentials file {string} with value: {string}")
```
Comment (Vipinofficial11) [inline src/e2e-test/java/io/cdap/plugin/salesforcebatchsource/stepsdesign/DesignTimeSteps.java]: You have to remove this method when you are raising the PR, same goes for the step. You can create this method in a different class for local testing but don't push that class.

File: src/e2e-test/java/io/cdap/plugin/salesforcesink/stepsdesign/DesignTimeSteps.java [hunk 15]
```diff
@@ -17,8 +17,36 @@
...
  */
 public class DesignTimeSteps {
+
+  @Then("Validate the values of records transferred to target Salesforce is equal to the values from source " +
+    "BigQuery table")
+  public void validateTheValuesOfRecordsTransferredToTargetSalesforceIsEqualToTheValuesFromSourceBigQueryTable()
+    throws IOException, InterruptedException, IOException, SQLException, ClassNotFoundException, ParseException {
+    int sourceBQRecordsCount = BigQueryClient.countBqQuery(PluginPropertyUtils.pluginProp("bqSourceTable"));
+    BeforeActions.scenario.write("No of Records from source BigQuery table:" + sourceBQRecordsCount);
+    Assert.assertEquals("Out records should match with target PostgreSQL table records count",
```
Comment (Vipinofficial11) [inline src/e2e-test/java/io/cdap/plugin/salesforcesink/stepsdesign/DesignTimeSteps.java]: Update the message Postgres is mentioned here.

File: src/e2e-test/java/io/cdap/plugin/tests/hooks/TestSetupHooks.java [hunk 16]
```diff
@@ -164,4 +170,44 @@ public static void deletePushTopic() {
...
+    } catch (BigQueryException e) {
+      if (e.getMessage().contains("Not found: Table")) {
+        BeforeActions.scenario.write("BQ Target Table " + bqTargetTableName + " does not exist");
+      } else {
+        Assert.fail(e.getMessage());
+      }
+    }
+  }
+
+  @After(order = 1, value = "@BQ_SINK_TEST")
```
Comment (Vipinofficial11) [inline src/e2e-test/java/io/cdap/plugin/tests/hooks/TestSetupHooks.java]: update the hook name as it is for multi source it can cause issues and confusion while running pipelines.

File: src/e2e-test/java/io/cdap/plugin/tests/hooks/TestSetupHooks.java [hunk 17]
```diff
@@ -164,4 +170,44 @@ public static void deletePushTopic() {
...
+    } catch (BigQueryException e) {
+      if (e.getMessage().contains("Not found: Table")) {
+        BeforeActions.scenario.write("BQ Target Table " + bqTargetTableName + " does not exist");
+      } else {
+        Assert.fail(e.getMessage());
+      }
+    }
+  }
+
+  @Before(order = 2, value = "@BQ_SINsK_TEST")
```
Comment (Vipinofficial11) [inline src/e2e-test/java/io/cdap/plugin/tests/hooks/TestSetupHooks.java]: Correct the name of hook it should be BQ_SINK_TEST

File: src/e2e-test/java/io/cdap/plugin/tests/hooks/TestSetupHooks.java [hunk 18]
```diff
@@ -123,6 +128,7 @@ private static void createSourceBQTableWithQueries(String bqCreateTableQueryFile
         , StandardCharsets.UTF_8);
       createTableQuery = createTableQuery.replace("DATASET", PluginPropertyUtils.pluginProp("dataset"))
         .replace("TABLE_NAME", bqSourceTable);
+      System.out.println(bqSourceTable);
```
Comment (Vipinofficial11) [inline src/e2e-test/java/io/cdap/plugin/tests/hooks/TestSetupHooks.java]: remove sout

File: src/e2e-test/java/io/cdap/plugin/tests/hooks/TestSetupHooks.java [hunk 19]
```diff
@@ -99,12 +103,13 @@ public static void createTempSourceBQTableForLeadSObject() throws IOException, I
     String company = uniqueId + ".com";
 
     BigQueryClient.getSoleQueryResult("create table `" + bqSourceDataset + "." + bqSourceTable + "` as " +
-      "SELECT * FROM UNNEST([ STRUCT('" + firstName + "' AS FirstName, '" + lastName + "' AS LastName, '"
-      + company + "' AS Company)])");
+                                        "SELECT * FROM UNNEST([ STRUCT('" + firstName + "' AS FirstName, '" +
```
Comment (Vipinofficial11) [inline src/e2e-test/java/io/cdap/plugin/tests/hooks/TestSetupHooks.java]: Correct the formating

File: src/e2e-test/java/io/cdap/plugin/tests/hooks/TestSetupHooks.java [hunk 20]
```diff
@@ -82,12 +86,12 @@ public static void deleteTemperoryCreatedBQTable() throws IOException, Interrupt
...
    * test data.
-   *
+   * <p>
    * Sample row:
    * FirstName | LastName | Company                               |
-   *     22    | 968      | 245308db-6088-4db2-a933-f0eea650846a  |
+   * 22    | 968      | 245308db-6088-4db2-a933-f0eea650846a  |
    */
-  @Before(order = 1, value = "@BQ_SOURCE_TEST")
+  @Before(order = 1, value = "@BQ_SOUxcRCE_TEST")
```
Comment (Vipinofficial11) [inline src/e2e-test/java/io/cdap/plugin/tests/hooks/TestSetupHooks.java]: Fix the naming

File: src/e2e-test/java/io/cdap/plugin/tests/hooks/TestSetupHooks.java [hunk 21]
```diff
@@ -82,12 +86,12 @@ public static void deleteTemperoryCreatedBQTable() throws IOException, Interrupt
   /**
    * Create BigQuery table with 3 columns (FirstName - string, LastName - string, Company - string) containing random
    * test data.
-   *
+   * <p>
```
Comment (Vipinofficial11) [inline src/e2e-test/java/io/cdap/plugin/tests/hooks/TestSetupHooks.java:101]: Why added a <p> tag here ?

File: src/e2e-test/java/io/cdap/plugin/tests/hooks/TestSetupHooks.java [hunk 22]
```diff
@@ -45,22 +45,26 @@
  */
 public class TestSetupHooks {
 
-  @Before(order = 2, value = "@BQ_SINK_TEST")
-  public void setTempTargetBQTable() {
-    String bqTargetTable = "TestSN_table" + RandomStringUtils.randomAlphanumeric(10);
-    PluginPropertyUtils.addPluginProp("bqTargetTable", bqTargetTable);
-    BeforeActions.scenario.write("BigQuery Target table name: " + bqTargetTable);
+  @Before(order = 1, value = "@CREATE_CUSTOM")
+  public static void createCustom() throws UnsupportedEncodingException, ParseException {
```
Comment (Vipinofficial11) [inline src/e2e-test/java/io/cdap/plugin/tests/hooks/TestSetupHooks.java]: Instead of writing @CREATE_CUSTOM use something @CREATE_TEST_DATA and @DELETE_TEST_DATA

File: src/e2e-test/java/io/cdap/plugin/tests/hooks/TestSetupHooks.java [hunk 23]
```diff
@@ -45,22 +45,26 @@
...
+    BeforeActions.scenario.write("Creating new Custom Object..");
+
+    JSONObject custom = new JSONObject(PluginPropertyUtils.pluginProp("CustomObjectData"));
```
Comment (Vipinofficial11) [inline src/e2e-test/java/io/cdap/plugin/tests/hooks/TestSetupHooks.java]: instead of naming it custom writ testData

File: src/e2e-test/java/io/cdap/plugin/tests/hooks/TestSetupHooks.java [hunk 24]
```diff
@@ -45,22 +45,26 @@
...
+    SalesforceClient.createObject(custom, SObjects.AUTOMATION_CUSTOM__C.value);
+
+  }
+
+  @After(order = 2, value = "@DELETE_OBJECT")
+  public static void deleteCustomObject() {
+    String id = SalesforceClient.id;
```
Comment (Vipinofficial11) [inline src/e2e-test/java/io/cdap/plugin/tests/hooks/TestSetupHooks.java]: id -> uniqueRecordId

File: src/e2e-test/java/io/cdap/plugin/utils/SalesforceClient.java [hunk 25]
```diff
@@ -71,9 +75,10 @@ public class SalesforceClient {
   private static final String REST_ENDPOINT = PluginPropertyUtils.pluginProp("rest.api.endpoint");
   private static final String API_VERSION = PluginPropertyUtils.pluginProp("rest.api.version");
   private static final Header prettyPrintHeader = new BasicHeader("X-PrettyPrint", "1");
+  public static List<JsonObject> objectResponseList = new ArrayList<>();
+  public static String id;
```
Comment (Vipinofficial11) [inline src/e2e-test/java/io/cdap/plugin/utils/SalesforceClient.java]: update the name here as well id -> uniqueRecordId

File: src/e2e-test/java/io/cdap/plugin/utils/SalesforceClient.java [hunk 26]
```diff
@@ -71,9 +75,10 @@ public class SalesforceClient {
   private static final String REST_ENDPOINT = PluginPropertyUtils.pluginProp("rest.api.endpoint");
   private static final String API_VERSION = PluginPropertyUtils.pluginProp("rest.api.version");
   private static final Header prettyPrintHeader = new BasicHeader("X-PrettyPrint", "1");
+  public static List<JsonObject> objectResponseList = new ArrayList<>();
```
Comment (Vipinofficial11) [inline src/e2e-test/java/io/cdap/plugin/utils/SalesforceClient.java]: objectResponseList -> sobjectResponse

File: src/e2e-test/java/io/cdap/plugin/utils/SalesforceClient.java [hunk 27]
```diff
@@ -176,11 +181,13 @@ public static void deletePushTopic(String pushTopicName) {
           PluginPropertyUtils.pluginProp("login.url"), 30000, "")));
 
       QueryResult queryResult = SalesforceStreamingSourceConfig.runQuery(partnerConnection,
-        String.format("SELECT Id FROM PushTopic WHERE Name = '%s'", pushTopicName));
+                                                                         String.format(
```
Comment (Vipinofficial11) [inline src/e2e-test/java/io/cdap/plugin/utils/SalesforceClient.java]: Fix the indentation

File: src/e2e-test/java/io/cdap/plugin/utils/SalesforceClient.java [hunk 28]
```diff
@@ -191,10 +198,74 @@ public static void deletePushTopic(String pushTopicName) {
...
+      } else {
+        // Handle other status codes or error scenarios
+        logger.info("Failed to delete Id.");
+      }
+    } catch (IOException ioException) {
+      logger.info("Error in establishing connection to Salesforce: " + ioException);
+    }
+  }
+
+  public static String queryObjectID(String objectName) {
```
Comment (Vipinofficial11) [inline src/e2e-test/java/io/cdap/plugin/utils/SalesforceClient.java]: Where is this method used.

File: src/e2e-test/resources/pluginParameters.properties [hunk 29]
```diff
@@ -114,3 +118,13 @@ account.schema= [{"key":"Id","value":"string"},{"key":"IsDeleted","value":"boole
   {"key":"JigsawCompanyId","value":"string"},{"key":"AccountSource","value":"string"},\
   {"key":"SicDesc","value":"string"}]
 
+CreateBQTableQueryFile=BigQuery/BigQueryCreateTableQuery.txt
+InsertBQDataQueryFile=BigQuery/BigQueryInsertDataQuery.txt
+
+bigQueryDatatypesColumnsList=(Id,Name,Col_Timestamp__c,Col_Date__c,Col_Currency__c,Col_Email__c,Col_Number__c,\
+  Col_GeoLocation__Latitude__s,Col_GeoLocation__Longitude__s,Col__c,Col_Url__c,Col_Time__c,Col_Text__c)
```
Comment (Vipinofficial11) [inline src/e2e-test/resources/pluginParameters.properties:127]: I see you are creating the data here as well as in the text files. I will suggest use a single method for data creation.

File: src/e2e-test/features/salesforcebatchsource/RunTime.feature (hunk 1 above)
Comment (AnkitCLI) [reply src/e2e-test/features/salesforcebatchsource/RunTime.feature]: done

File: src/e2e-test/java/io/cdap/plugin/BQValidation.java (hunk 5 above)
Comment (AnkitCLI) [reply src/e2e-test/java/io/cdap/plugin/BQValidation.java]: done

File: src/e2e-test/java/io/cdap/plugin/BQValidation.java (hunk 6 above)
Comment (AnkitCLI) [reply src/e2e-test/java/io/cdap/plugin/BQValidation.java]: done

File: src/e2e-test/java/io/cdap/plugin/BQValidation.java (hunk 7 above)
Comment (AnkitCLI) [reply src/e2e-test/java/io/cdap/plugin/BQValidation.java]: done

File: src/e2e-test/java/io/cdap/plugin/BQValidation.java (hunk 9 above)
Comment (AnkitCLI) [reply src/e2e-test/java/io/cdap/plugin/BQValidation.java:157]: done

File: src/e2e-test/java/io/cdap/plugin/BQValidation.java (hunk 10 above)
Comment (AnkitCLI) [reply src/e2e-test/java/io/cdap/plugin/BQValidation.java]: done

File: src/e2e-test/java/io/cdap/plugin/package-info.java (hunk 12 above)
Comment (AnkitCLI) [reply src/e2e-test/java/io/cdap/plugin/package-info.java:2]: This file is used to remove checkstyle errors and the time of build

File: src/e2e-test/java/io/cdap/plugin/salesforcebatchsource/stepsdesign/DesignTimeSteps.java (hunk 14 above)
Comment (AnkitCLI) [reply src/e2e-test/java/io/cdap/plugin/salesforcebatchsource/stepsdesign/DesignTimeSteps.java]: done

File: src/e2e-test/java/io/cdap/plugin/salesforcesink/stepsdesign/DesignTimeSteps.java (hunk 15 above)
Comment (AnkitCLI) [reply src/e2e-test/java/io/cdap/plugin/salesforcesink/stepsdesign/DesignTimeSteps.java]: done

File: src/e2e-test/java/io/cdap/plugin/tests/hooks/TestSetupHooks.java (hunk 16 above)
Comment (AnkitCLI) [reply src/e2e-test/java/io/cdap/plugin/tests/hooks/TestSetupHooks.java]: done

File: src/e2e-test/java/io/cdap/plugin/tests/hooks/TestSetupHooks.java (hunk 17 above)
Comment (AnkitCLI) [reply src/e2e-test/java/io/cdap/plugin/tests/hooks/TestSetupHooks.java]: done

File: src/e2e-test/java/io/cdap/plugin/tests/hooks/TestSetupHooks.java (hunk 18 above)
Comment (AnkitCLI) [reply src/e2e-test/java/io/cdap/plugin/tests/hooks/TestSetupHooks.java]: done

File: src/e2e-test/java/io/cdap/plugin/tests/hooks/TestSetupHooks.java (hunk 20 above)
Comment (AnkitCLI) [reply src/e2e-test/java/io/cdap/plugin/tests/hooks/TestSetupHooks.java]: done

File: src/e2e-test/java/io/cdap/plugin/tests/hooks/TestSetupHooks.java (hunk 21 above)
Comment (AnkitCLI) [reply src/e2e-test/java/io/cdap/plugin/tests/hooks/TestSetupHooks.java:101]: removed done

File: src/e2e-test/java/io/cdap/plugin/tests/hooks/TestSetupHooks.java (hunk 22 above)
Comment (AnkitCLI) [reply src/e2e-test/java/io/cdap/plugin/tests/hooks/TestSetupHooks.java]: done

File: src/e2e-test/java/io/cdap/plugin/tests/hooks/TestSetupHooks.java (hunk 23 above)
Comment (AnkitCLI) [reply src/e2e-test/java/io/cdap/plugin/tests/hooks/TestSetupHooks.java]: done

File: src/e2e-test/java/io/cdap/plugin/tests/hooks/TestSetupHooks.java (hunk 24 above)
Comment (AnkitCLI) [reply src/e2e-test/java/io/cdap/plugin/tests/hooks/TestSetupHooks.java]: done

File: src/e2e-test/java/io/cdap/plugin/utils/SalesforceClient.java (hunk 25 above)
Comment (AnkitCLI) [reply src/e2e-test/java/io/cdap/plugin/utils/SalesforceClient.java]: done

File: src/e2e-test/java/io/cdap/plugin/utils/SalesforceClient.java (hunk 26 above)
Comment (AnkitCLI) [reply src/e2e-test/java/io/cdap/plugin/utils/SalesforceClient.java]: done

File: src/e2e-test/java/io/cdap/plugin/utils/SalesforceClient.java (hunk 28 above)
Comment (AnkitCLI) [reply src/e2e-test/java/io/cdap/plugin/utils/SalesforceClient.java]: This method is used in multiObject validation method.

File: src/e2e-test/features/salesforcebatchsource/RunTime.feature (hunk 2 above)
Comment (AnkitCLI) [reply src/e2e-test/features/salesforcebatchsource/RunTime.feature]: done

File: src/e2e-test/features/salesforcemultiobjectsbatchsource/RunTime.feature (hunk 3 above)
Comment (AnkitCLI) [reply src/e2e-test/features/salesforcemultiobjectsbatchsource/RunTime.feature]: done

File: src/e2e-test/features/salesforcesink/RunTime.feature (hunk 4 above)
Comment (AnkitCLI) [reply src/e2e-test/features/salesforcesink/RunTime.feature]: done

File: src/e2e-test/java/io/cdap/plugin/tests/hooks/TestSetupHooks.java (hunk 19 above)
Comment (AnkitCLI) [reply src/e2e-test/java/io/cdap/plugin/tests/hooks/TestSetupHooks.java]: done

File: src/e2e-test/java/io/cdap/plugin/BQValidation.java (hunk 11 above)
Comment (AnkitCLI) [reply src/e2e-test/java/io/cdap/plugin/BQValidation.java]: done

File: src/e2e-test/java/io/cdap/plugin/utils/SalesforceClient.java (hunk 27 above)
Comment (AnkitCLI) [reply src/e2e-test/java/io/cdap/plugin/utils/SalesforceClient.java]: done

File: src/e2e-test/java/io/cdap/plugin/salesforcebatchsource/stepsdesign/DesignTimeSteps.java (hunk 13 above)
Comment (AnkitCLI) [reply src/e2e-test/java/io/cdap/plugin/salesforcebatchsource/stepsdesign/DesignTimeSteps.java]: done
